test/tree-sitter-corpus.jl.faceup

```diff
@@ -0,0 +1,784 @@
+
+«k:function» «f:f»(x, y::«t:Int», «:julia-ts-quoted-symbol-face:z»=«c:1», ws...) «k:end»
+
+«k:function» (::«t:Type{Int}», x::«t:Int» = «c:1», y::«t:Int»...) «k:end»
+
+«k:function» «f:apply»(f, args...; kwargs...)
+«k:end»
+
+«k:function» «f:g»(; x, y::«t:Int», «:julia-ts-quoted-symbol-face:z» = «c:1», kwargs...) «b:nothing» «k:end»
+
+«x:# ==================================================»
+«x:# function definition return types»
+«x:# ==================================================»
+
+«k:function» «f:s»(n)::«t:MyInt»
+    MyInt(n + «c:1»)
+«k:end»
+
+«k:function» «f:bar»(f, xs::Foo.«t:Bar»)::Foo.«t:Bar»
+    map(f, xs)
+«k:end»
+
+«x:# ==================================================»
+«x:# function definition tuple parameters»
+«x:# ==================================================»
+
+«k:function» «f:swap»((x, y))
+    (y, x)
+«k:end»
+
+«k:function» «f:f»((x, y)=(«c:1»,«c:2»))
+    (x, y)
+«k:end»
+
+«k:function» «f:car»((x, y)::«t:Tuple»{«t:T», «t:T»}) «k:where» «t:T»
+    x
+«k:end»
+
+«x:# ==================================================»
+«x:# type parametric function definition parameters»
+«x:# ==================================================»
+
+«k:function» «f:f»(x::«t:T») «k:where» «t:T»
+«k:end»
+
+«k:function» «f:f»(n::«t:N») «k:where» {«t:N» <: «t:Integer»}
+    n
+«k:end»
+
+f(n::«t:N», m::«t:M») «k:where» {«t:N» <: «t:Number»} «k:where» {«t:M» <: «t:Integer»} = n^m
+
+«t:Foo»{«t:T»}(x::«t:T») «k:where» {«t:T»} = x
+
+«k:function» «f:norm»(p::«t:Point»{«t:T»} «k:where» T<:Real)
+    norm2(p)
+«k:end»
+
+Base.«f:show»(io::«t:IO», ::«t:MIME"text/plain"», m::«t:Method»; kwargs...) = show_method(io, m, kwargs)
+
+«x:# ==============================»
+«x:# macro definitions»
+«x:# ==============================»
+
+«k:macro» «f:name»(s::«t:Symbol»)
+    String(s)
+«k:end»
+
+«k:macro» «f:count»(args...) length(args) «k:end»
+
+«x:# ==============================»
+«x:# identifiers»
+«x:# ==============================»
+
+abc_123_ABC
+_fn!
+ρ; φ; z
+ℝ
+x′
+θ̄
+logŷ
+ϵ
+ŷ
+🙋
+🦀
+
+«x:# ==============================»
+«x:# field expressions»
+«x:# ==============================»
+
+foo.x
+bar.x.y.z
+
+(a[«c:1»].b().c).d
+
+Base.«:julia-ts-quoted-symbol-face::+»
+
+df.«s:"a"»
+
+«x:# ==============================»
+«x:# index expressions»
+«x:# ==============================»
+
+a[«c:1», «c:2», «c:3»]
+a[«c:1», :]
+«s:"foo"»[«c:1»]
+
+«x:# ==============================»
+«x:# type parametrized expressions»
+«x:# ==============================»
+
+«t:Vector»{«t:Int»}
+«t:Vector»{<:«t:Number»}
+«:julia-ts-interpolation-expression-face:$(»«D:usertype»«:julia-ts-interpolation-expression-face:)»{«t:T»}
+
+{«:julia-ts-quoted-symbol-face::x»} ~ normal(«c:0», «c:1»)
+
+«x:# ==============================»
+«x:# function call expressions»
+«x:# ==============================»
+
+f()
+g(«s:"hi"», «c:2»)
+h(d...)
+
+f(e; «:julia-ts-quoted-symbol-face:f» = g)
+g(arg; kwarg)
+
+«k:new»{typeof(xs)}(xs)
```

Comment:
`new` is not a keyword here.

Reply:
I added a query deliberately to make it a fontified as a keyword.

Technically, you are right, it is not a reserved word, and can be assigned. From a practical point of view, it should be treated as a reserved word inside a constructor definition. I personally like that it gets highlighted as a keyword, because it stands out better, and helps me, as a developer, avoid the accidental overwriting of its value.

WDYT?

Reply:
I'd lean towards font-locking being descriptive rather than proscriptive, and I can think of a lot of situations where a variable named "new" is what makes the most sense for a local variable.

Reply:
Fair enough, I removed this query.